analyzer-core/src/main/java/ffb/analyzer/core/pubsub/Publisher.java

```diff
@@ -0,0 +1,79 @@
+package ffb.analyzer.core.pubsub;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class Publisher {
+    private Object mutex = new Object();
+
+    public enum Event {
+        EXAMPLE_EVENT
+    }
+
+    private static Publisher publisher;
+    private Map<Event, Set<Subscriber>> subscribers = new HashMap<>();
```

Comment:
I think we want a `List<Subscriber>` here. If a subscriber wants to listen to two different events of the same type, it can't with a `Set`.

Reply:
On second thought, maybe I'm misunderstanding how you envision `Subscribers`. In my mind, each `Subscriber` is a different process that needs to be notified on the results. So if my browser kicks off 3 events (may or may not be the same type), I'd have three different result notifications. Is that the same as what you thought?